presto-common/src/main/java/com/facebook/presto/common/PrestoVersion.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PrestoVersion
+{
+    private static final Pattern VERSION_REGEX = Pattern.compile("^(0\\.([1-9][0-9]*))(\\.([1-9][0-9]*))?(\\.*)?(-([2][0-9]{7}\\.[0-2][0-9]{5}-[0-9]{1,}|SNAPSHOT))?$");
+    private int majorVersion;
+    private int minorVersion;
+    private String snapshotVersion;
+
+    public PrestoVersion(String version)
+    {
+        Matcher matcher = VERSION_REGEX.matcher(version);
+        this.majorVersion = Integer.parseInt(Optional.ofNullable(matcher.lookingAt() ? matcher.group(2) : null).orElse("0"));
+        this.minorVersion = Integer.parseInt(Optional.ofNullable(matcher.lookingAt() ? matcher.group(4) : null).orElse("0"));
+        this.snapshotVersion = Optional.ofNullable(matcher.lookingAt() ? matcher.group(7) : null).orElse("");
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return this.majorVersion * 1000 + this.minorVersion;
+    }
+
+    @Override
+    public boolean equals(Object version)
+    {
+        if (this == version) {
+            return true;
+        }
+        try {
+            PrestoVersion prestoVersion = (PrestoVersion) version;
+            return (this.majorVersion == prestoVersion.majorVersion
+                    && this.minorVersion == prestoVersion.minorVersion
+                    && ((this.snapshotVersion.isEmpty()
+                    && prestoVersion.snapshotVersion.isEmpty())
+                    || this.snapshotVersion.equals(prestoVersion.snapshotVersion)
+                    || this.snapshotVersion.equals("SNAPSHOT")
+                    || prestoVersion.snapshotVersion.equals("SNAPSHOT")));
+        }
+        catch (Exception e) {
+            throw new IllegalArgumentException("Please compare with a PrestoVersion object");
+        }
+    }
+
+    public boolean equalTo(PrestoVersion prestoVersion)
+    {
+        return this.equals(prestoVersion);
+    }
+
+    public boolean lessThan(PrestoVersion version)
```

Comment:
You can override `compareTo` method which returns a negative integer, zero, or a positive integer as this object is less than, equal to, or greater than the specified object
and make rest of the methods call into `compareTo`